Havit.Blazor.Components.Web.Bootstrap/wwwroot/HxModal.js

```diff
@@ -26,6 +27,21 @@ function handleModalShown(event) {
 	event.target.hxModalDotnetObjectReference.invokeMethodAsync('HxModal_HandleModalShown');
 };
 
+async function handleModalHide(event) {
+    let modalInstance = bootstrap.Modal.getInstance(event.target);
+
+    if (modalInstance.dontPrevent)
+        return;
+
+    event.preventDefault();
+
+    let cancel = await event.target.hxModalDotnetObjectReference.invokeMethodAsync('HxModal_HandleModalHide');
+    if (!cancel) {
+        modalInstance.dontPrevent = true;
```

Comment:
1. Is there any chance the `modelInstance` will be reused and the `dontPrevent` remains `true`? Should we set the attribute to `false` somewhere (for sure)?

2. I would go for some name which will contain the `hide` event name, such as `hidePreventionDisabled`.

Reply:
1. As long as `hidden.bs.modal` event is triggered, modal instance is being disposed (inside `handleModalHidden` event handler). I tried, but couldn't get into a situation where modal instance is reused. 
2. Changed.
3. Accidentally I've seen an incorrect event listener removal. Fixed in 0527282.

Reply:
To accept such feature to the library I think we need the fast-path to be available for the major use case where the `hide` event is not intercepted. (As such intercept not only sends the signal to Blazor but also relatively aggressively influences the Bootstrap hiding workflow  - cancels it and "restores" it only if confirmed by Blazor-code).

The only way to implement it without incorporating any additional complex logic or new parameters is to rely on `OnHiding.HasDelegate` and sacrifice the `virtual InvokeOnHidingAsync()` seam allowing further logic in derived components.

Would you please add such logic to your PR? The main point is "if there is no `OnHiding` callback subscribed, do not subscribe to the `hide` event of the underlying Bootstrap component et al". (Will need some related logic to cover the conditional un-subscription in JS. You don't need to support the scenario where the `OnHiding` callback is set after the component is rendered for the first time.)

Reply:
Conditional subscription to the "hide" event: ed3aee3

I didn't add conditional un-subscription, because of
> Calling removeEventListener() with arguments that do not identify any currently registered event listener on the EventTarget has no effect.

from https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener

Reply:
We still need to remove the `protected virtual Task InvokeOnHidingAsync()` as we do not want to allow confusion for the authors of derived components to spend their time investigating why their `override InvokeOnHidingAsync()` is not being called (when the `OnHiding` parameter handler is not set). Just call directly `OnHiding.InvokeAsync()` from `HandleModalHide()`.

Reply:
Done. 8b6bb17